spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/authentication/OAuth2AuthenticationDetails.java

```diff
@@ -49,16 +48,19 @@
 	/**
 	 * Records the access token value and remote address and will also set the session Id if a session already exists
 	 * (it won't create one).
-	 * 
+	 *
 	 * @param request that the authentication request was received from
 	 */
 	public OAuth2AuthenticationDetails(HttpServletRequest request) {
-		this.tokenValue = (String) request.getAttribute(ACCESS_TOKEN_VALUE);
-		this.tokenType = (String) request.getAttribute(ACCESS_TOKEN_TYPE);
-		this.remoteAddress = request.getRemoteAddr();
+		this(request.getRemoteAddr(), request.getSession(false) != null ? request.getSession(false).getId() : null,
+				(String) request.getAttribute(ACCESS_TOKEN_TYPE), (String) request.getAttribute(ACCESS_TOKEN_VALUE));
+	}
 
-		HttpSession session = request.getSession(false);
-		this.sessionId = (session != null) ? session.getId() : null;
+	public OAuth2AuthenticationDetails(String remoteAddress, String sessionId, String tokenType, String tokenValue) {
```

Comment:
`OAuth2AuthenticationDetails` was intended to be used with `HttpServletRequest`, so I don't see a reason to provide this overloaded constructor. If you require it, then you can simply `extend OAuth2AuthenticationDetails` and provide an overloaded constructor just the same.